imports/plugins/core/orders/client/containers/ordersListContainer.js

```diff
@@ -140,30 +162,347 @@ class OrdersListContainer extends Component {
+        whichFalseState = shippingStates.picked;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isNotPacked++;
+        whichFalseState = shippingStates.packed;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/packed") {
+        isNotLabeled++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/labeled") {
+        isLabeled++;
+      } else {
+        // check if the selected order(s) are being regressed back to this state
+        if (order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/labeled") || order.shipping[0].workflow.status === "coreOrderWorkflow/shipped") {
+          ordersToRegress++;
+        }
+      }
+    });
+
+    // display regression alert if order(s) are being regressed
+    if (ordersToRegress) {
+      this.displayRegressionAlert(selectedOrders, ordersToRegress, status, whichFalseState, isNotPacked, isNotLabeled, isLabeled);
+
+      // display proper alert if the order(s) are in this state already or want to skip the previous states
+    } else this.displayAlert(selectedOrders, status, whichFalseState, isNotPacked, isNotLabeled, isLabeled);
+  }
+
+
+  shippedShippingStatus = (selectedOrders, status) => {
+    let isNotLabeled = 0;
+    let isNotShipped = 0;
+    let isShipped = 0;
+    let whichFalseState = "";
+
+    selectedOrders.forEach((order) => {
+      // check if the order(s) are in this state already or in one of the previous states
+      if (order.shipping[0].workflow.status === "new") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.picked;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.packed;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/packed") {
+        isNotLabeled++;
+        whichFalseState = shippingStates.labeled;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/labeled") {
+        isNotShipped++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/shipped") {
+        isShipped++;
+      }
+    });
+
+    // display proper alert if the order(s) are in this state already or want to skip the previous states
+    this.displayAlert(selectedOrders, status, whichFalseState, isNotLabeled, isNotShipped, isShipped);
+  }
+
+  /**
+   * setShippingStatus
+   *
+   * @summary call the relevant method based on the provided shipping status
+   * @param {String} status - the selected shipping status to be set
+   * @param {Array} selectedOrdersIds - array of ids of the selected orders
+   * @return {null} no return value
+   */
+  setShippingStatus = (status, selectedOrdersIds) => {
+    this.setState({
+      renderFlowList: true
+    });
+    const selectedOrders = this.state.orders.filter((order) => {
+      return selectedOrdersIds.includes(order._id);
+    });
+
+    if (status === "picked") {
```

Comment:
I wonder if there was a way to make this dynamic so we didn't need a conditional for each type which feels brittle. In the interest of getting this done I would be okay with the implementation as it is though.